router.go

```diff
@@ -288,15 +291,19 @@ func (r *Router) ServeHTTP(w http.ResponseWriter, req *http.Request) {
 		defer r.recv(w, req)
 	}
 
-	var path string
-	if r.CleanPath && req.URL.Path != "*" {
-		path = CleanPath(req.URL.EscapedPath())
-	} else {
-		path = req.URL.EscapedPath()
+	path := req.URL.Path
+	if r.UseRawPath && len(req.URL.RawPath) > 0 {
+		path = req.URL.RawPath
+	}
+
+	// This isn't ideal bc we're secretly matching against a different URL
+	// than what we'll pass on to the handler but it solves our existing use
+	// case and for the most part it will be ok.
+	if r.CleanPath {
```

Comment:
was the `req.URL.Path != "*"` check superfluous?

Reply:
no it was not. great catch, thanks